apps/explorer/src/hooks/useTokenList.ts

```diff
@@ -0,0 +1,54 @@
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import useSWR from 'swr'
+import { SWR_NO_REFRESH_OPTIONS } from '@cowprotocol/common-const'
+import type { TokenInfo, TokenList } from '@uniswap/token-lists'
+
+type TokenListByAddress = Record<string, TokenInfo>
+type TokenListPerNetwork = Record<SupportedChainId, TokenListByAddress>
+
+const INITIAL_TOKEN_LIST_PER_NETWORK: TokenListPerNetwork = {
+  [SupportedChainId.MAINNET]: {},
+  [SupportedChainId.GNOSIS_CHAIN]: {},
+  [SupportedChainId.SEPOLIA]: {},
+}
+
+export function useTokenList(chainId: SupportedChainId | undefined): { data: TokenListByAddress; isLoading: boolean } {
+  const { data: cowSwapList, isLoading: isCowListLoading } = useTokenListByUrl(
+    'https://files.cow.fi/tokens/CowSwap.json'
+  )
+  const { data: coingeckoList, isLoading: isCoingeckoListLoading } = useTokenListByUrl(
+    'https://tokens.coingecko.com/uniswap/all.json'
+  )
+  const { data: honeyswapList, isLoading: isHoneyswapListLoading } = useTokenListByUrl(
+    chainId === SupportedChainId.GNOSIS_CHAIN ? 'https://tokens.honeyswap.org' : ''
+  )
```

Comment:
Using only these 3 for now, as this should be enough for most cases.
Coingecko has quite a lot of tokens, cowswap is just so we can use our own lists/images, honeyswap is for gchain only.